### PR TITLE
Add load_earth_geoid function for Earth Geoid dataset

### PR DESCRIPTION
**Pull request by willschlitzer**
This adds the `load_earth_geoid` function to load the EGM2008 Earth Geoid remote dataset.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/earth_geoid.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        Grid registration type. Either ``"pixel"`` for pixel registration or
+        ``"gridline"`` for gridline registration. Default is ``None``, where
+        a pixel-registered grid is returned unless only the
+        gridline-registered grid is available.
+
+    Returns
+    -------
+    grid : :class:`xarray.DataArray`
+        The Earth geoid grid. Coordinates are latitude and
+        longitude in degrees. Units are in milligals (mGal).
```
[the remark this answers is not in the document]
Comment (willschlitzer) [reply pygmt/datasets/earth_geoid.py:53]: Okay; I was under the impression it was mGal, but I've never worked with this dataset.

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
@willschlitzer Need to resolve the conflicts after merging #2196

**Comment by willschlitzer** [comment]
I just pushed a change for the magnetic anomaly cache files, as there were previously no commas after  `earth_mag_01d_g.grd` and `S90W180.earth_mag_05m_g.grd`. It didn't cause any problems in  #2196 (as far as I can tell), but should I rerun caching in this to make sure the appropriate files have been cached?

**Comment by seisman** [comment]
> I just pushed a change for the magnetic anomaly cache files, as there were previously no commas after `earth_mag_01d_g.grd` and `S90W180.earth_mag_05m_g.grd`. It didn't cause any problems in #2196 (as far as I can tell), but should I rerun caching in this to make sure the appropriate files have been cached?

I think it's fine.

**Pull request merged**